lib/compress/zstd_cwksp.h

```diff
@@ -272,7 +276,7 @@ MEM_STATIC size_t ZSTD_cwksp_bytes_to_align_ptr(void* ptr, const size_t alignByt
  * which we can allocate from the end of the workspace.
  */
 MEM_STATIC void*  ZSTD_cwksp_initialAllocStart(ZSTD_cwksp* ws) {
-    return (void*)((size_t)ws->workspaceEnd & ~(ZSTD_CWKSP_ALIGNMENT_BYTES-1));
+    return (void*)((size_t)ws->workspaceEnd & (size_t)~(ZSTD_CWKSP_ALIGNMENT_BYTES-1));
```

Comment:
I don't think this cast is actually correct. You need to cast inside the not expression if your goal is to ensure that the not doesn't happen to a narrower type that then gets zero-extended during promotion to the `size_t`.

Also should we be casting to `intptr_t`, not `size_t`?

Reply:
> Also should we be casting to `intptr_t`, not `size_t`?

Unfortunately, `intptr_t` is not guaranteed to exist, while `size_t` is.

Also, since this is just for alignment checks, we just care about the last bits (realistically, the last 6 bits maximum for cache-line alignment). So we could even use a smaller type. But `size_t` is good enough already.

Reply:
rewrote `ZSTD_cwksp_initialAllocStart()` for clarity